### PR TITLE
Show rayline when cursor click

### PR DESCRIPTION
**Pull request by PennyQ**
I implement the ray cursor line into Glue 3D Viewers as shown below, using the current picking icon on the menu bar.

But due to the normalization of the volume data, the transform of those `ray markers` are not correct along the ray casting line :(  Could you help me to have a look @astrofrog ? Many thanks!

image

### CODE REVIEW

File: glue_vispy_viewers/volume/volume_toolbar.py [hunk 1]
```diff
@@ -9,27 +9,83 @@
...
+                inter_value.append(data_array[each_pos])
+
+            max_data_value = inter_points_index[np.argmax(inter_value), :3]
+            colors[np.argmax(inter_value)] = (1, 1, 0.5, 1)  # change the color of max value marker
+
+            self.markers.set_data(pos=inter_points_index, face_color=colors)
+            print('interpos is', inter_points_index, data_array)
+            # update limit to make transform sync with the stretch  :)
+            self._vispy_widget.add_data_visual(self.markers)
+            self._vispy_widget._update_limits()
```
Comment (PennyQ) [inline glue_vispy_viewers/volume/volume_toolbar.py]: Here is the line that I tried to set the transform of `Markers/ray line` to the normalized which makes it sync when doing stretch on UI control. But it doesn't work as I expected :(

**Comment by astrofrog** [comment]
@PennyQ - I'll take a look first thing tomorrow morning!

**Comment by astrofrog** [comment]
@PennyQ - can you rebase this too? (just to make sure I have the latest version with the bundled vispy before doing any debugging)

**Comment by PennyQ** [comment]
@astrofrog done :)

File: glue_vispy_viewers/volume/volume_toolbar.py [hunk 2]
```diff
@@ -9,27 +9,83 @@
 import numpy as np
 from matplotlib import path
 from glue.core.roi import RectangularROI, CircularROI, PolygonalROI
-
+from vispy import scene
```
Comment (astrofrog) [inline glue_vispy_viewers/volume/volume_toolbar.py]: @PennyQ: this should now be:

``` python
from ..bundled.vispy import scene
```

Comment (astrofrog) [inline glue_vispy_viewers/volume/volume_toolbar.py]:
```
from ..extern.vispy import scene
```

**Comment by PennyQ** [comment]
I add a ray line with a marker to represent the max value, the figure is what it looks like:
image

Todo: 
- [x]  show position on the status bar
- [ ]  implement floodfill selection
- [ ] implement dendrogram slelection (may need a loading scheme for dendrogram file)

@astrofrog The frame line of gridbox becomes 'wider' when I did the selection, is this normal or not? (you can see from the figure above)

**Comment by astrofrog** [comment]
> is this normal or not?

No, this shouldn't happen - not sure what's going on there!

**Pull request closed without merging**